cmd/bucket-handlers-listobjects.go

```diff
@@ -100,7 +102,8 @@ func (api objectAPIHandlers) ListObjectsV2Handler(w http.ResponseWriter, r *http
 	}
 
 	for i := range listObjectsV2Info.Objects {
-		if listObjectsV2Info.Objects[i].IsEncrypted() {
+		if crypto.S3.IsEncrypted(listObjectsV2Info.Objects[i].UserDefined) ||
```

Comment:
You should use `crypto.IsEncrypted(...)` here. `S3/SSEC.IsEncrypted()` may not detect tampered objects.